src/backend/joanie/tests/core/test_api_products.py

```diff
@@ -0,0 +1,287 @@
+"""Test suite for the Product API"""
+from joanie.core import enums, factories, models
+from joanie.tests.base import BaseAPITestCase
+
+
+class ProductApiTest(BaseAPITestCase):
+    """Test the API of the Product resource."""
+
+    def test_api_product_read_list_anonymous(self):
+        """
+        It should not be possible to retrieve the list of products for anonymous users.
+        """
+        factories.ProductFactory()
+
+        response = self.client.get("/api/products/")
+
+        self.assertContains(
+            response,
+            "The requested resource was not found on this server.",
+            status_code=404,
+        )
+
+    def test_api_product_read_list_authenticated(self):
+        """
+        It should not be possible to retrieve the list of products for authenticated users.
+        """
+        factories.ProductFactory()
+        user = factories.UserFactory.build()
+        token = self.get_user_token(user.username)
+
+        response = self.client.get(
+            "/api/products/", HTTP_AUTHORIZATION=f"Bearer {token}"
+        )
+
+        self.assertContains(
+            response,
+            "The requested resource was not found on this server.",
+            status_code=404,
+        )
+
+    def test_api_product_read_detail(self):
+        """
+        Any users should be allowed to retrieve a product with a minimal db access.
+        """
+        product = factories.ProductFactory(type=enums.PRODUCT_TYPE_CREDENTIAL)
```

Comment:
```suggestion
        product = factories.ProductFactory(type=enums.PRODUCT_TYPE_CREDENTIAL)
```
why product type credential, the test doesn't explain why

Reply:
To have a certificate